.circleci/config.yml

```diff
@@ -367,82 +190,75 @@ jobs:
             nix-env -iA cachix -f https://cachix.org/api/v1/install
             cachix use postgrest
       - run:
-          name: Build the PostgREST statically linked binary
-          command: |
-            nix-build -A postgrestStatic
+          name: Install testing scripts for all supported PostgreSQL versions
+          command: nix-env -f default.nix -iA tests
+      # Our utility scripts use cabal to run the tests, which will pick up all
+      # the libraries that are built with Nix. So this should only ever build
+      # the PostgREST itself and be reasonably quick. We accelerate that part
+      # further by caching `~/.cabal` and `dist-newstyle`.
+      - restore_cache:
+          keys:
+            - nix-test-{{ checksum "postgrest.cabal" }}-{{ checksum "nix/nixpkgs-version.nix" }}
+            - nix-test-{{ checksum "postgrest.cabal" }}
+            - nix-test
```

Comment:
These ones slipped me by. Are they correct? They look redundant.

Reply:
@steve-chavez Ah, yes, we missed to discuss that. These additional keys should help to make the test runs a bit faster using the partial cache restore feature of circleci (https://circleci.com/docs/2.0/caching/#restoring-cache). The keys are prefix matched in order. If the more  specific key is not found, it will try the next one and take the newest cache that matches the prefix.

In line 226 we append the current time to the caching key, so a new cache will always be saved (it should never become very large, as all dependencies are pre built by Nix). If the nix pin or cabal file changes, the build will not have to start completely from scratch, it will use the latest cache which is hopefully still a bit similar and accelerare the process.

Reply:
@monacoremo Ah, got it. I didn't know that one. Also missed the `epoch` cache key.

Could you add those explanations as comments on the circle config? Maybe on #1538.

Reply:
Ah, got it. I didn't know that one. Also missed the epoch cache key. Thanks for the explanation!

Could you add comments with the explanations on the circle config? Maybe on #1538.